once_test.py

```diff
@@ -729,6 +814,29 @@ async def counting_fn2():
         self.assertEqual(await counting_fn2(), 2)
         self.assertEqual(await counting_fn2(), 2)
 
+    async def test_once_per_thread(self):
+        counter = Counter()
+
+        @once.once(per_thread=True)
+        async def counting_fn(*args) -> int:
+            """Returns the call count, which should always be 1."""
+            nonlocal counter
+            del args
+            return counter.get_incremented()
+
+        @execute_with_barrier(n_workers=_N_WORKERS)
```

Comment:
ditto, you want the barrier inside the counting function, not outside

Reply:
Made it into a sandwich!

Reply:
And this barrier needs a comment that it's just for racing